modules/opensearch/opensearch.install

```diff
@@ -20,10 +20,34 @@ function opensearch_migrate_settings() {
     'ting_recommendation_url' => 'opensearch_recommendation_url',
     'ting_enable_logging' => 'opensearch_enable_logging',
     'ting_cache_lifetime' => 'opensearch_cache_lifetime',
+    'ting_ranking_custom' => 'opensearch_ranking_custom',
+    'ting_ranking_fields' => 'opensearch_ranking_fields',
+    'ting_sort_default' => 'opensearch_sort_default',
+    'ting_boost_fields' => 'opensearch_boost_fields',
   );
 
   array_walk($settings_map, function($new_var, $old_var) {
     variable_set($new_var, variable_get($old_var));
     variable_del($old_var);
   });
 }
+
+/**
+ * Translate legacy ting roles to opensearch.
+ */
+function migrate_roles() {
```

Comment:
This function never seems to be called?

Also: It seems as if a function for migrating permissions could be pretty useful going forward. I thought about implementing a generic version in the `ding_base` module but that would introduce a range of indirect dependencies which we may not be interested in...